### PR TITLE
Scope functions

### PR DESCRIPTION
**Pull request by p7nov**
1. Added missing scope functions: `run`, `apply`, `also`.
2. Renamed chapter _06_stdlib_ to _06_scope_functions_.
3. Renamed files _01_letFunction.md_ -> _01_let.md_ and _02_withFunction.md_ -> _03_with.md_ for consistency.
4. Revised existing texts.

### CODE REVIEW

File: examples/06_scope_functions/04_apply.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+```kotlin
+data class Person(var name: String, var age: Int, var about: String) {
+    constructor() : this("", 0, "")
+}
+
+fun main() {
+//sampleStart
+    val jake = Person()                                     // 1
+    val stringDescription = jake.apply {                    // 2
+        name = "jake"                                       // 3
```
Comment (AlexanderPrendota) [inline examples/06_scope_functions/04_apply.md]: Jake. The first letter should be in upper case

File: examples/06_scope_functions/05_also.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+             constructor() : this("", 0, "")
+}
+         
+fun logCreation(p: Person) {
+    println("A new person ${p.name} was is created.")              
+}
+         
+fun main() {
+//sampleStart
+    val jake = Person("jake", 30, "Android developer")   // 1
```
Comment (AlexanderPrendota) [inline examples/06_scope_functions/05_also.md]: Jake. The first letter should be in upper case

File: examples/06_scope_functions/01_let.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+<div class="language-kotlin" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+var i = 0
+fun funWithSideEffect(): String = ".".repeat(i++)
+
+fun main() {
+//sampleStart
+    val r = funWithSideEffect().let {                              // 1
```
Comment (AlexanderPrendota) [inline examples/06_scope_functions/01_let.md]: I guess the variable should have the normal name instead of one "random" letter

Comment (p7nov) [reply examples/06_scope_functions/01_let.md]: Rewrote the whole example

File: examples/06_scope_functions/01_let.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    printNonNull("my string") 
+//sampleEnd
+}
+```
+
+</div>
+
+1. Calls the given block on the result on the string "_test_".
+2. Calls the function on "_test_" by the `it` reference.
+3. `let` returns value of this expression.
```
Comment (B1ggDave) [inline examples/06_scope_functions/01_let.md]: returns the value of this expression

File: examples/06_scope_functions/02_run.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+//sampleEnd
+}
+```
+
+</div>
+
+
+1. Calls the given block on a nullable variable.
+2. Inside `run`, the object's members are accessed without its name.
+3. The functional block returns the `lenght` of the given `String` if it's not `null`.
```
Comment (B1ggDave) [inline examples/06_scope_functions/02_run.md]: `length`

File: examples/06_scope_functions/03_with.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
+
+
 # with
 
-*with* function lets you access members of an object in a short and concise way. Inside the `with` block you can refer to object members without specifying its name as a prefix. 
+`with` is a non-extension function that can be used to access members of its argument in a concise way: you can omit the instance when referring to its members.
```
Comment (B1ggDave) [inline examples/06_scope_functions/03_with.md]: `with` is a non-extension function that can access members of its argument concisely: you can omit the instance when referring to its members.

File: examples/06_scope_functions/04_apply.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# apply
+
+`apply` executes a given block on an object and returns the object itself. Inside the block, the object is referenced by `this`.
```
Comment (B1ggDave) [inline examples/06_scope_functions/04_apply.md]: `apply` executes a block on an object and returns the object itself. Inside the block, the object is referenced by `this`.

File: examples/06_scope_functions/04_apply.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+//sampleEnd
+    println(stringDescription)
+}
+```
+
+</div>
+
+
+1. Creates a `Person()` instance with default property values.
+2. Applies the given functional block (next 3 lines) to the instance.
```
Comment (B1ggDave) [inline examples/06_scope_functions/04_apply.md]: Applies the functional block (next 3 lines) to the instance.

File: examples/06_scope_functions/05_also.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# also
+
+`also` works quite like [`apply`](04_apply.md): it executes a given block and returns the called object.
+Inside the block, the object is referenced by `it`, so it's easier to pass it as an agrument.
+This function is handy for embedding side effects, such as logging, in call chains.
```
Comment (B1ggDave) [inline examples/06_scope_functions/05_also.md]: This function is handy for embedding side effects, such as logging in call chains. 

This function is handy for embedding side effects, such as logging, and in call chains. 


'embedding side effects' -- side effects are normally something negative.
alts:
side benefits
secondary effects

File: examples/06_scope_functions/05_also.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        .also {                                          // 2 
+            writeCreationLog(it)                         // 3
+         }
+         //sampleEnd
+        }
+```
+
+</div>
+
+1. Creates a `Person()` object with given property values.
```
Comment (B1ggDave) [inline examples/06_scope_functions/05_also.md]: 1. Creates a `Person()` object with the given property values.

File: examples/06_scope_functions/01_let.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
+# let
+
+`let` is a Kotlin standard library function that can be used for scoping and null-checks. When called on an object, `let` executes the given functional block and returns the result of its last expression.
```
Comment (B1ggDave) [inline examples/06_scope_functions/01_let.md]: [only a suggestion]

The Kotlin standard library function 'let' can be used for scoping and null-checks.

The Kotlin standard library includes the function 'let' which can be used for scoping and null-checks.

**Pull request merged**